### ISSUE TITLE
Runtime error UnboundLocalError: local variable 'should_play_default' referenced before assignment

### ISSUE THREAD

**Issue by thebertster** [im85288/service.upnext#140]
In playbackmanager.py, show_popup_and_wait() may exit early if player has already stopped by the time the function is called. This causes showing_next_up_page and showing_still_watching_page to both be returned as False.

Subsequently, extract_play_info() is called but this currently assumes that either the next_up_page or still_watching_page has been displayed and will give a runtime error when it hits `return should_play_default, should_play_non_default` as neither of these variables has been set. This situation arises in real use if I start an episode and then jump forward to the end of playback for example. This just needs to be handled with:

```
    def extract_play_info(self, next_up_page, showing_next_up_page, showing_still_watching_page, still_watching_page):
        if showing_next_up_page:
.....
        elif showing_still_watching_page:
.....
        else:
            return False, False

```

**Comment by dagwieers** [comment]
Personally I would like to redesign this part of the code, but we need integration tests first to ensure that code changes result in expected or deterministic behaviour. But I am contemplating writing an add-on that would run prepared tests in real Kodi using Up Next.

**Comment by dagwieers** [comment]
In the meantime we do accept pull requests that improve this code for known deficiencies ;-)

**Comment by thebertster** [comment]
PR submitted.

**Comment by dagwieers** [comment]
Great thanks!

**Comment by dagwieers** [comment]
Closed by #142
